include/Game/Entities/Diode.h

```diff
@@ -30,14 +33,40 @@ struct Obj : public Plants::Obj {
 struct Mgr : public EnemyMgrBaseAlwaysMovieActor {
 	Mgr(int, u8);
 
-	virtual ~Mgr();                                     // _58 (weak)
-	virtual EnemyBase* birth(EnemyBirthArg&);           // _70
-	virtual void createObj(int);                        // _A0 (weak)
-	virtual EnemyBase* getEnemy(int);                   // _A4 (weak)
-	virtual void doAlloc();                             // _A8
-	virtual EnemyTypeID::EEnemyTypeID getEnemyTypeID(); // _AC (weak)
-	virtual void loadModelData();                       // _C8 (weak)
-	virtual J3DModelData* doLoadBmd(void*);             // _D4 (weak)
+	virtual ~Mgr() { }                        // _58 (weak)
+	virtual EnemyBase* birth(EnemyBirthArg&); // _70
+	virtual void createObj(int count)         // _A0 (weak)
+	{
+		m_obj = new Obj[count];
+	}
+	virtual EnemyBase* getEnemy(int idx) // _A4 (weak)
+	{
+		return &m_obj[idx];
+	}
+	virtual void doAlloc();                            // _A8
+	virtual EnemyTypeID::EEnemyTypeID getEnemyTypeID() // _AC (weak)
+	{
+		return EnemyTypeID::EnemyID_DaiodoGreen;
+	}
+	virtual void loadModelData() // _C8 (weak)
+	{
+		EnemyMgrBase::loadModelData();
+		if (m_modelData == nullptr) {
+			JUTException::panic_f("plantsMgr.h", 578, "P2Assert");
```

Comment:
it would - however, I really dislike the idea of having every plant-related struct in the same single header when we have the other entities split out so nicely, so I made it a direct exception call. It'd be possible to put it in a single header if we really wanted to be super duper consistent though

Reply:
though I suppose you could include the various plant headers in a master plantMgr header if you really wanted.

Reply:
is it infeasible to include the various plant headers in plantMgr.h?

Reply:
I mean the diode-specific weak functions would have to get declared in plantMgr.h not their individual headers to make the exception work, so I think it'd be a bit of an odd setup, but it's technically an option I guess?

Reply:
wouldn’t that just have to be defined rather than declared in that header?

Reply:
I'm not sure if defining the weak functions away from the declaration will mess with the generation order or not, and I'm hesitant to mess with restrict our options in fixing that tbqh